### PR TITLE
[Merged by Bors] - Add merge transition docs

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Add some documentation about migrating pre-merge Lighthouse to post-merge Lighthouse.

## Additional Info

NA

### CODE REVIEW

**Comment by paulhauner** [comment]
I've marked this as `ready-for-review` since I'm keen to make it available. I'm interested in typos and corrections where things are wrong, but I'd prefer to put any additions in a new PR in the spirit of something being better than nothing. Future PRs can always expand upon it.

File: book/src/merge-migration.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+contract](https://ethereum.org/en/staking/deposit-contract/) and beacon chain proposers include this
+information in beacon blocks in order to on-board new validators. BNs exclusively use the `eth`
+namespace on the eth1 [JSON-RPC API](https://ethereum.org/en/developers/docs/apis/json-rpc/) to
+achieve this.
+
+To progress through the Bellatrix upgrade nodes will need a *new* connection to an "eth1" node;
+`--execution-endpoint`. This connection has a few different properties. Firstly, the term "eth1
+node" has been deprecated and replaced with "execution engine". Whilst "eth1 node" and "execution
+engine" still refer to the same projects (Geth, Nethermind, etc) the former refers to the pre-merge
+versions and the later refers to post-merge versions. Secondly, there is a strict one-to-one
```
Comment (pawanjay176) [inline book/src/merge-migration.md]:
```suggestion
versions and the latter refers to post-merge versions. Secondly, there is a strict one-to-one
```

**Review by pawanjay176 — APPROVED** [review]
LGTM apart from the tiny typo

**Pull request closed without merging**